### PR TITLE
One refactor for Executor operations from JdbcExecutorLoader class

### PR DESCRIPTION
**Pull request by kunkun-tang**
This code patch is a follow-up of #1345. Similar to that PR, I start moving executor related operations. The first step is to create a delegate class and move all related methods to the new class. 

Note that **I didn't change anything, but only move methods from one place to another place**. After this PR, I will change the implementation inside, and add unit test. The benefits of this PR is to help team do review.

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/executor/ExecutorDBManager.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+ * the License.
+ */
+
+package azkaban.executor;
+
+import azkaban.database.AbstractJdbcLoader;
+import azkaban.db.DatabaseOperator;
+import azkaban.metrics.CommonMetrics;
+import azkaban.utils.Props;
+import com.google.inject.Singleton;
```
Comment (suvodeep-pyne) [inline azkaban-common/src/main/java/azkaban/executor/ExecutorDBManager.java]: As per @HappyRay suggestion, please use `javax.inject package`. check #1333

File: azkaban-common/src/main/java/azkaban/executor/ExecutorDBManager.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.inject.Inject;
+import org.apache.commons.dbutils.QueryRunner;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.apache.log4j.Logger;
+
+@Singleton
+public class ExecutorDBManager extends AbstractJdbcLoader {
```
Comment (suvodeep-pyne) [inline azkaban-common/src/main/java/azkaban/executor/ExecutorDBManager.java]: This pattern is pretty much a DAO pattern. Should we rename it to `ExecutorDao` then?

https://stackoverflow.com/questions/19154202/data-access-object-dao-in-java
https://en.wikipedia.org/wiki/Data_access_object

**Comment by suvodeep-pyne** [comment]
Another thing is to have tests for all the `DAO` methods. This can be a future thing but it's important since this is the class doing the actual DB work / has the core business logic.

File: azkaban-common/src/main/java/azkaban/executor/ExecutorDBManager.java (hunk 2 above)
Comment (kunkun-tang) [inline azkaban-common/src/main/java/azkaban/executor/ExecutorDBManager.java]: That's a good suggestion. I was considering using this name, but I didn't adopt it as I thought we don't have explicit `Executor` class, which sounded necessary for a DAO pattern to me. Now that there's no objection, I can rename it to DAO. Likely we could add and use Executor Class in future refactor.

File: azkaban-common/src/main/java/azkaban/executor/ExecutorDBManager.java (hunk 1 above)
Comment (kunkun-tang) [inline azkaban-common/src/main/java/azkaban/executor/ExecutorDBManager.java]: Modifying.

File: azkaban-common/src/main/java/azkaban/executor/ExecutorDBManager.java (hunk 2 above)
Comment (suvodeep-pyne) [inline azkaban-common/src/main/java/azkaban/executor/ExecutorDBManager.java]: Well, there is `azkaban.executor.Executor`. However, this is just a nitpick. I leave this to you.

This change looks good to go. I like the idea of moving the core data access layer to `azkaban-db`.

File: azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java [hunk 3]
```diff
@@ -32,14 +31,14 @@
 import org.apache.commons.dbutils.ResultSetHandler;
 import org.apache.log4j.Logger;
 
-@Singleton
-public class ExecutionFlowDBManager {
+@javax.inject.Singleton
```
Comment (suvodeep-pyne) [inline azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java]: why not import `javax.inject.Singleton` instead of using the entire path?

Comment (kunkun-tang) [inline azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java]: Good finding! I also noticed just now and am modifying.

**Comment by HappyRay** [comment]
Could you run saveAction and run CheckStyle on new files?

**Pull request merged**